react/features/base/settings/reducer.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+
+import { APP_WILL_MOUNT } from '../../app';
+import JitsiMeetJS, { browser } from '../lib-jitsi-meet';
+import { ReducerRegistry } from '../redux';
+import { PersistenceRegistry } from '../storage';
+import { randomHexString } from '../util';
+
+import { SETTINGS_UPDATED } from './actionTypes';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * The default/initial redux state of the feature {@code base/settings}.
+ *
+ * @type Object
+ */
+const DEFAULT_STATE = {};
+
+const STORE_NAME = 'features/base/settings';
+
+/**
+ * Sets up the persistence of the feature {@code base/settings}.
+ *
+ * Note: There are some values inherited from the old Settings.js that needs
+ * session-based storage but not persisted storage, such as cameraDeviceId
+ * and micDeviceId so we utilize the selective storage functionality of the
+ * persistency engine.
+ */
+PersistenceRegistry.register(STORE_NAME, {
+    audioOutputDeviceId: true,
+    avatarID: true,
+    avatarURL: true,
+    cameraDeviceId: false,
+    displayName: true,
+    email: true,
+    localFlipX: true,
+    micDeviceId: false,
+    serverURL: true,
+    startAudioOnly: true,
+    startWithAudioMuted: true,
+    startWithVideoMuted: true
+});
+
+ReducerRegistry.register(STORE_NAME, (state = DEFAULT_STATE, action) => {
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        return _initSettings(state);
+
+    case SETTINGS_UPDATED:
+        return {
+            ...state,
+            ...action.settings
+        };
+    }
+
+    return state;
+});
+
+/**
+ * Retrieves the legacy profile values regardless of it's being in pre or
+ * post-flattening format.
+ *
+ * @private
+ * @returns {Object}
+ */
+function _getLegacyProfile() {
+    let persistedProfile
+        = window.localStorage.getItem('features/base/profile');
```

Comment:
@lyubomir How important is backwards compatibility at this stage? If we are to keep it, when would be able to kill this?

Reply:
Related: I think we should give these legacy support code fragments a lifetime (e.g. 3 months) and kill them off after it passes. Any suggestions?

Reply:
Agreed. Can you add a FIXME note please?